lib/charms/tls_certificates_interface/v1/tls_certificates.py

```diff
@@ -239,10 +244,32 @@ def _on_certificate_revoked(self, event: CertificateRevokedEvent) -> None:
         replicas_relation.data[self.app].pop("chain")
         self.unit.status = WaitingStatus("Waiting for new certificate")
 
+    def _on_certificate_invalidated(self, event: CertificateInvalidatedEvent) -> None:
+        replicas_relation = self.model.get_relation("replicas")
+        if not replicas_relation:
+            self.unit.status = WaitingStatus("Waiting for peer relation to be created")
+            event.defer()
+            return
+        if event.reason == "revoked":
+            self._certificate_revoked()
+        if event.reason == "expired":
+            self._on_certificate_expiring(event)
```

Comment:
This doc should probably say
```suggestion
            self._certificate_expired(event)
```
because `on_cert_expiring` takes an event of a different type: `CertificateExpiringEvent`.

Reply:
It was previously extended to also take `CertificateInvalidatedEvent` like so:
```python
def _on_certificate_expiring(
        self, event: Union[CertificateExpiringEvent, CertificateInvalidatedEvent]
    ) -> None:
```
I've used that method because I think we also want to request a new certificate. Let me know if I understand it properly.